src/Language/PureScript/CodeGen/JS/AST.hs

```diff
@@ -126,120 +130,190 @@ data JS
   -- |
   -- A numeric literal
   --
-  = JSNumericLiteral (Either Integer Double)
+  = JSNumericLiteral (Maybe SourceSpan) (Either Integer Double)
```

Comment:
You mean `JS a = JSNumericLiteral a (Either Integer Double)` ? I guess that was my original thought, the type does appear everywhere so maybe an alias `type JS = JS1 (Maybe SourceSpan)` ?